Segmentation fault for unknown deepcopy-gen:interfaces package

When using deepcopy-gen and specify and unknown package in `//+k8s:deepcopy-gen:interfaces=<package>.<Interface>`, gengo see faults.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x12a3d6d]

goroutine 1 [running]:
k8s.io/gengo/parser.(*Builder).AddDirTo(0xc00007b590, 0xc004684ea0, 0x23, 0xc005eba548, 0xc007ee390c, 0x6)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/parser/parse.go:272 +0xad
k8s.io/gengo/generator.(*Context).AddDir(...)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/generator/generator.go:246
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).deepCopyableInterfacesInner(0xc004739f80, 0xc005eba540, 0xc004a08540, 0x1198ff2, 0xc0096ff4d0, 0x0, 0x0, 0x0)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/examples/deepcopy-gen/generators/deepcopy.go:533 +0x165
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).deepCopyableInterfaces(0xc004739f80, 0xc005eba540, 0xc004a08540, 0x0, 0x0, 0xc00467c500, 0x1, 0xc00467c500, 0xc00504d170)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/examples/deepcopy-gen/generators/deepcopy.go:550 +0x5d
k8s.io/gengo/examples/deepcopy-gen/generators.(*genDeepCopy).GenerateType(0xc004739f80, 0xc005eba540, 0xc004a08540, 0x13c7940, 0xc0046869e0, 0x0, 0x7ffeefbff9c6)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/examples/deepcopy-gen/generators/deepcopy.go:635 +0x21c
k8s.io/gengo/generator.(*Context).executeBody(0xc005eba540, 0x13c73e0, 0xc00013f0b0, 0x13d0098, 0xc004739f80, 0x2c0, 0x13783a8)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/generator/execute.go:306 +0x122
k8s.io/gengo/generator.(*Context).ExecutePackage(0xc00012f9d0, 0x7ffeefbffa30, 0xf, 0x13ce610, 0xc003e046c0, 0x0, 0x0)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/generator/execute.go:267 +0xb98
k8s.io/gengo/generator.(*Context).ExecutePackages(0xc00012f9d0, 0x7ffeefbffa30, 0xf, 0xc00504cb70, 0x1, 0x1, 0x0, 0x28)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/generator/execute.go:51 +0xc8
k8s.io/gengo/args.(*GeneratorArgs).Execute(0xc00011b220, 0xc0096ffe38, 0x1350639, 0x6, 0x137e610, 0x0, 0x0)
        /Users/rtf/go/pkg/mod/k8s.io/gengo@v0.0.0-20201214224949-b6c5ce23f027/args/args.go:207 +0x1fa
main.main()
        /Users/rtf/Github/fischor/labs-k8s-indexer-informer/vendor/k8s.io/code-generator/cmd/deepcopy-gen/main.go:77 +0x4cb
```

The relevant code 

https://github.com/kubernetes/gengo/blob/485abfe95c7c792c41929dea3578927f90485a9f/parser/parse.go#L266-L273

does not check if `b.buildPackages[dir]` exists.

Checking for existence, returning an error with an appropriate error message and displaying that to the user would be useful. Took me some time to realise that I had a typo in the `//+k8s:deepcopy-gen:interfaces=<package>.<Interface>` package.

--- a/examples/deepcopy-gen/generators/deepcopy.go
+++ b/examples/deepcopy-gen/generators/deepcopy.go
@@ -530,7 +530,10 @@ func (g *genDeepCopy) deepCopyableInterfacesInner(c *generator.Context, t *types
 	var ts []*types.Type
 	for _, intf := range intfs {
 		t := types.ParseFullyQualifiedName(intf)
-		c.AddDir(t.Package)
+		err := c.AddDir(t.Package)
+		if err != nil {
+			return nil, err
+		}
 		intfT := c.Universe.Type(t)
 		if intfT == nil {
 			return nil, fmt.Errorf("unknown type %q in %s tag of type %s", intf, interfacesTagName, intfT)

--- a/parser/parse.go
+++ b/parser/parse.go
@@ -269,7 +269,11 @@ func (b *Builder) AddDirTo(dir string, u *types.Universe) error {
 	if _, err := b.importPackage(dir, true); err != nil {
 		return err
 	}
-	return b.findTypesIn(canonicalizeImportPath(b.buildPackages[dir].ImportPath), u)
+	pkg, ok := b.buildPackages[dir]
+	if !ok {
+		return fmt.Errorf("no such package: %q", dir)
+	}
+	return b.findTypesIn(canonicalizeImportPath(pkg.ImportPath), u)
 }
 
 // AddDirectoryTo adds an entire directory to a given Universe. Unlike AddDir,
@@ -283,7 +287,11 @@ func (b *Builder) AddDirectoryTo(dir string, u *types.Universe) (*types.Package,
 	if _, err := b.importPackage(dir, true); err != nil {
 		return nil, err
 	}
-	path := canonicalizeImportPath(b.buildPackages[dir].ImportPath)
+	pkg, ok := b.buildPackages[dir]
+	if !ok {
+		return nil, fmt.Errorf("no such package: %q", dir)
+	}
+	path := canonicalizeImportPath(pkg.ImportPath)
 	if err := b.findTypesIn(path, u); err != nil {
 		return nil, err
 	}

